src/PROPOSAL/PROPOSAL/crosssection/CrossSectionDNDX/CrossSectionDNDXInterpolant.h

```diff
@@ -11,44 +11,52 @@
 
 namespace PROPOSAL {
 
-double transform_relative_loss(double v_cut, double v_max, double v);
-double retransform_relative_loss(double v_cut, double v_max, double v);
+double transform_relative_loss(double v_cut, double v_max, double v, double c = 1);
+double retransform_relative_loss(double v_cut, double v_max, double v, double c = 1);
 double transform_loss_log(double v_cut, double v_max, double v);
 double retransform_loss_log(double v_cut, double v_max, double v);
 
-template <typename Param> struct transform_loss {
-    static std::function<double(double, double, double)> func;
-};
+namespace crosssection {
+    struct Compton;
+    struct Ionization;
+}
 
-template <typename Param> struct retransform_loss {
-    static std::function<double(double, double, double)> func;
-};
+// general transformation rules for v nodes
+// TODO: in C++17, this section can be rewritten much cleaner using if constexpr
 
-template <typename Param>
-std::function<double(double, double, double)> transform_loss<Param>::func
-    = [](double v_cut, double v_max, double v) {
-          return transform_relative_loss(v_cut, v_max, v);
-      };
+template <typename Param, std::enable_if_t<!(std::is_base_of<crosssection::Ionization, Param>::value || std::is_base_of<crosssection::Compton, Param>::value), bool> = true>
+double transform_loss(double v_cut, double v_max, double v) {
+    return transform_relative_loss(v_cut, v_max, v);
+}
 
-template <typename Param>
-std::function<double(double, double, double)> retransform_loss<Param>::func
-    = [](double v_cut, double v_max, double v) {
-          return retransform_relative_loss(v_cut, v_max, v);
-      };
+template <typename Param, std::enable_if_t<!(std::is_base_of<crosssection::Ionization, Param>::value || std::is_base_of<crosssection::Compton, Param>::value), bool> = true>
+double retransform_loss(double v_cut, double v_max, double v) {
+    return retransform_relative_loss(v_cut, v_max, v);
+}
 
-namespace crosssection {
-    struct ComptonKleinNishina;
+// specifications for Ionization
+
+template <typename Param, std::enable_if_t<std::is_base_of<crosssection::Ionization, Param>::value, bool> = true>
+double transform_loss(double v_cut, double v_max, double v) {
+    return transform_relative_loss(v_cut, v_max, v, 1.5);
```

Comment:
Can you explain, why this additional exponentiation of 1.5 during transformation decreases the interpolation error?

Reply:
This is what the distribution of interpolation nodes look like with c=1 (so the default distribution):

![plot3_master](https://user-images.githubusercontent.com/15159319/148681203-64685e9a-de85-4be7-a4c9-38468d333c39.png)

Note that I plotted only every fifth node here so that it is easier to see the effect. 

And this is what the distribution of the interpolation nodes look like with c=1.5:

![plot3_master](https://user-images.githubusercontent.com/15159319/148681248-7cb4ab45-b0e8-452c-8b1a-6ca25cf32d2a.png)

There are more nodes for small v now, which is the kinematic range where the interpolated value is actually changing.